src/mobile-v3/lib/src/app/dashboard/repository/user_preferences_repository.dart

```diff
@@ -46,13 +42,16 @@ class UserPreferencesImpl extends UserPreferencesRepository with NetworkLoggy {
   }
 
   @override
-  Future<Map<String, dynamic>> getUserPreferences(String userId,
-      {String? groupId}) async {
-    final String url =
-        '$apiBaseUrl/users/preferences/$userId${groupId != null ? "?group_id=$groupId" : ""}';
+  Future<Map<String, dynamic>> getUserPreferences(String userId, {String? groupId}) async {
+    String url = '$apiBaseUrl$preferencesEndpoint/$userId';
+    
+    if (groupId != null && groupId.isNotEmpty) {
```

Comment:
For anything to do with the Group ID, please utilise the  default AirQo Group ID while at it. As in `group_id=64f54e4621d9b90013925a08`

This can be put in an environmental variable accordingly.